### PR TITLE
Add accuracy func and test mnist accuracy

### PR DESCRIPTION
**Pull request by kexinzhao**
The accuracy is tested on the mnist test set.

### CODE REVIEW

File: src/example/mnist/test_mnist.cc [hunk 1]
```diff
@@ -35,35 +39,41 @@ bool is_file_exist(const std::string& fileName) {
...
-                 "file doesn't exist; have you run create_mnist_recordio.py");
-  auto reader = CreateRecordioFileReader(
-      filename, {32, 1, 28, 28, 32, 1}, {4, 2}, {0, 0});
-
-  Linear linear1(784, 200, "relu");
-  Linear linear2(200, 200, "relu");
-  Linear linear3(200, 10, "relu");
+  std::string filename1 = "/tmp/mnist_train.recordio";
+  std::string filename2 = "/tmp/mnist_test.recordio";
+  PADDLE_ENFORCE(is_file_exist(filename1),
```
Comment (tonyyang-svail) [inline src/example/mnist/test_mnist.cc]: Maybe put `is_file_exist` inside `CreateRecordioFileReader`

File: src/function.h [hunk 2]
```diff
@@ -110,6 +110,9 @@ class Linear {
                             {{"X", {pre_bias}}, {"Y", {b_}}},
                             {{"Out", {pre_act}}},
                             add_op_attrs);
+    if (act_.empty()) {
```
Comment (tonyyang-svail) [inline src/function.h:114]: Maybe also make `Conv` support `act_.empty()`

File: src/function.h [hunk 3]
```diff
@@ -397,15 +423,19 @@ VariableHandle CreateRecordioFileReader(std::string filename,
...
   reader->GetMutable<paddle::framework::ReaderHolder>()->ReadNext(&data_holder);
   if (data_holder.empty()) {
     reader->GetMutable<paddle::framework::ReaderHolder>()->ReInit();
-    reader->GetMutable<paddle::framework::ReaderHolder>()->ReadNext(
-        &data_holder);
+    if (repeat) {
+      reader->GetMutable<paddle::framework::ReaderHolder>()->ReadNext(
+          &data_holder);
+    } else {
+      return {};
```
Comment (tonyyang-svail) [inline src/function.h]: Will this `return {}` hide the check `PADDLE_ENFORCE(!data_holder.empty(), "Error reading file.");`?

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

File: src/function.h (hunk 3 above)
Comment (kexinzhao) [reply src/function.h]: Good point. Revised.

File: src/function.h (hunk 2 above)
Comment (kexinzhao) [reply src/function.h:114]: Done for both conv and batchnorm

File: src/example/mnist/test_mnist.cc (hunk 1 above)
Comment (kexinzhao) [reply src/example/mnist/test_mnist.cc]: Done

**Review by kexinzhao — COMMENTED** [review]
Comments addressed.

**Review by tonyyang-svail — APPROVED** [review]
Excellent!

**Pull request merged**